tests/unittests/GraphOptzTest.cpp

```diff
@@ -2371,6 +2371,58 @@ TEST_F(GraphOptz, VarsCSE) {
   EXPECT_TRUE(input3->getUsers().begin()->getUser() == RN);
 }
 
+TEST_F(GraphOptz, VarsCSENaN) {
+  // Create two variables that are Private, are not trainable, have no writers
+  // and include NaNs. The first two variables have the same data, and so should
+  // be combined via variable CSE.  In particular, the NaN constants should not
+  // prevent the variables from being combine.
+  auto *input1 = mod_.createConstant(ElemKind::FloatTy, {5}, "input1");
+  auto *input2 = mod_.createConstant(ElemKind::FloatTy, {5}, "input2");
+  auto nan = std::numeric_limits<float>::quiet_NaN();
+  input1->getHandle() = {nan, nan, nan, nan, nan};
+  input2->getHandle() = {nan, nan, nan, nan, nan};
```

Comment:
I'd add some variety NAN + non NAN values in the constants to make sure it's handled properly.